### ISSUE TITLE
Markdown preview of HTTP with JSON body turns green

### ISSUE THREAD

**Issue by gurksaften** [microsoft/vscode#207443]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.87.1
- OS Version: Windows 10 Pro 22H2 64-bit

Steps to Reproduce:

1. Create a `.md` file with the following contents:
````
```http
HTTP/1.1 200 OK
Content-Type: application/json

{
	"ab": "a for abc abc a"
}
```
````
2. Open the preview of the the file and it will render the JSON strings as italic green (i assume comment-style).
![image](https://github.com/microsoft/vscode/assets/163146473/2f9a5f14-2dd1-408d-ad35-394a6fdc6f2b)

3. Add a space between `f` and `or` in the word `for` in the body causes the rendering of the JSON to return to normal.
![image](https://github.com/microsoft/vscode/assets/163146473/92ac0a88-4bd2-45cc-a7d5-0da685bdb080)

This seems to affect the entire body when the bug triggers causing all JSON strings to turn italic green unless you find the offending part of the JSON.

**Comment by Gobinath-B** [comment]
@mjbvz i want to work on this issue

**Comment by mjbvz** [comment]
This is an issue with highlight js we use in the preview. Switching to use VS Code's highlighting seems like the proper fix long term: #91279

**Comment by gurksaften** [comment]
Good job, @mjbvz, thank you. 👍
